docs/en/04_Changelogs/4.0.0.md

```diff
@@ -1095,6 +1095,55 @@ Some methods on `Requirements` have had their method signatures changed:
 A new config `Requirements_Backend.combine_in_dev` has been added in order to allow combined files to be
 forced on during development. If this is off, combined files is only enabled in live environments.
 
+Form validation has been refactored significantly. A new `FormMessage` trait has been created to
+handle field-level and form-level messages. This has the following properties:
+
+* `setMessage` to assign a message, type, and cast
+* `getMessage` retrieves the message string
+* `getMessageType` retrieves the message type (E.g. error, good, info)
+* `getMessageCast` retrieves the cast type
+* `getMessageCastingHelper` retrieves the DBField cast to use for the appropriate message cast
+* `getSchemaMessage` encodes this message for form schema use in ReactJS.
+
+`Form` methods have been changed:
+
+* `validate` method now returns a `ValidationResult` instance. This is no longer automatically
```

Comment:
Perhaps we want to:

 * Create a 2nd method with a new name. `getValidatonResult()`?
 * Rewrite `validate()` as a wrapper around this method that returns a boolean and throws a deprecation?